dashboard/src/components/AppView/SecretsTable/SecretItem.tsx

```diff
@@ -1,71 +1,70 @@
+  name: string;
+  secret?: IKubeItem<ISecret>;
+  getSecret: () => void;
 }
 
-interface ISecretItemState {
-  showSecret: { [s: string]: boolean };
-}
-
-class SecretItem extends React.Component<ISecretItemProps, ISecretItemState> {
-  public constructor(props: ISecretItemProps) {
-    super(props);
-    const showSecret = {};
-    if (this.props.secret.data) {
-      Object.keys(this.props.secret.data).forEach(k => (showSecret[k] = false));
-    }
-    this.state = { showSecret };
+class SecretItem extends React.Component<ISecretItemProps> {
+  public componentDidMount() {
+    this.props.getSecret();
   }
 
   public render() {
-    const { secret } = this.props;
-    const secretEntries: JSX.Element[] = [];
-    if (!_.isEmpty(this.props.secret.data)) {
-      Object.keys(secret.data).forEach(k => {
-        secretEntries.push(this.renderSecretEntry(k));
-      });
-    } else {
-      secretEntries.push(<span key="empty">The secret is empty</span>);
-    }
+    const { name, secret } = this.props;
     return (
       <tr className="flex">
-        <td className="col-2">{secret.metadata.name}</td>
-        <td className="col-2">{secret.type}</td>
-        <td className="col-7 padding-small">{secretEntries}</td>
+        <td className="col-3">{name}</td>
+        {this.renderSecretInfo(secret)}
       </tr>
     );
   }
 
-  private renderSecretEntry = (name: string) => {
-    const toggle = () => this.toggleDisplay(name);
-    const text = atob(this.props.secret.data[name]);
-    return (
-      <span key={name} className="flex">
-        <a onClick={toggle}>{this.state.showSecret[name] ? <EyeOff /> : <Eye />}</a>
-        <span className="flex margin-l-normal">
-          <span>{name}:</span>
-          {this.state.showSecret[name] ? (
-            <pre className="SecretContainer">
-              <code className="SecretContent">{text}</code>
-            </pre>
-          ) : (
-            <span className="margin-l-small">{text.length} bytes</span>
-          )}
-        </span>
-      </span>
-    );
-  };
-
-  private toggleDisplay = (name: string) => {
-    const { showSecret } = this.state;
-    this.setState({ showSecret: { ...showSecret, [name]: !showSecret[name] } });
-  };
+  private renderSecretInfo(secret?: IKubeItem<ISecret>) {
+    if (secret === undefined || secret.isFetching) {
+      return (
+        <td className="col-9">
+          <LoadingWrapper type={LoaderType.Placeholder} />
+        </td>
+      );
+    }
+    if (secret.error) {
+      return (
+        <td className="col-9">
+          <span className="flex">
+            <AlertTriangle />
+            <span className="flex margin-l-normal">Error: {secret.error.message}</span>
+          </span>
+        </td>
+      );
+    }
+    if (secret.item) {
+      const item = secret.item;
+      return (
+        <React.Fragment>
+          <td className="col-2">{item.type}</td>
+          <td className="col-7 padding-small">
+            {item.data ? (
```

Comment:
we should also check if `data` is an empty object (that's why `isEmpty` was being used)

Reply:
good point, however I noticed that `data` is never defined in the API response from an empty Secret. Just in case though, I'll use `_.isEmpty` here and add a comment

Reply:
I know, this is just in case that behavior changes. Thanks